src/colorizer/IRenderCanvas.ts

```diff
@@ -0,0 +1,68 @@
+import { Vector2 } from "three";
+
+import { ViewerStoreState } from "../state/slices";
+
+const canvasStateDeps = [
+  "dataset",
+  "featureKey",
+  "track",
+  "showTrackPath",
+  "colorRamp",
+  "colorRampRange",
+  "categoricalPalette",
+  "outlineColor",
+  "outlierDrawSettings",
+  "outOfRangeDrawSettings",
+  "inRangeLUT",
+  "vectorMotionDeltas",
+  "vectorVisible",
+  "vectorColor",
+  "vectorScaleFactor",
+  "backdropKey",
+  "backdropVisible",
+  "objectOpacity",
+  "backdropSaturation",
+  "backdropBrightness",
+] as const;
+
+export type RenderCanvasStateParams = Pick<ViewerStoreState, (typeof canvasStateDeps)[number]>;
+
+export const renderCanvasStateParamsSelector = (state: ViewerStoreState): RenderCanvasStateParams => {
+  const entries = canvasStateDeps.map((key) => [key, state[key]]);
+  return Object.fromEntries(entries);
+};
+
+/**
+ * A canvas that renders timelapse data.
+ */
+export interface IRenderCanvas {
+  get domElement(): HTMLCanvasElement;
+  /** (X,Y) resolution of the canvas, in pixels. */
+  get resolution(): Vector2;
+  setResolution(width: number, height: number): void;
+  /**
+   * Updates the parameters used to configure the canvas view. See
+   * `CanvasStateParams` for a complete list of parameters required.
+   */
+  setParams(params: RenderCanvasStateParams): void;
```

Comment:
This is the biggest change, that will most affect `ColorizeCanvas`. I removed the individual property setters and joined the logic into one function call.

Before, there were many individual setters for properties in `ColorizeCanvas`, and many of them had additional side-effects, like triggering a re-render.

This change makes it so that all the parameter changes can be handled at once, and parameters can be pulled directly out of state. Rendering only takes place once, after all the properties have been updated.